sympy/series/series.py

```diff
@@ -61,3 +62,110 @@ def series(expr, x=None, x0=0, n=6, dir="+"):
     """
     expr = sympify(expr)
     return expr.series(x, x0, n, dir)
+
+
+def inversion_series(f, x, a=0, n=3):
+    r"""
+    The Lagrange inversion theorem (or Lagrange inversion formula, which we abbreviate
+    as LIT), also known as the Lagrange--Bürmann formula, gives the Taylor series
+    expansion of the inverse function of an analytic function.
+    It finds important applications in combinatorics and calculus.
+
+    The theorem was proved by Lagrange (1736--1813) and generalized by the German
+    mathematician and teacher Hans Heinrich Bürmann ( --1817), both in the late 18th
+    century.
+
+    Theorem: Suppose x is defined as a function of y by an equation of the form
+
+    f(x) = y
+
+    where f is analytic at a point a and f'(a) is not equal to 0.
+    Then it is possible to invert or solve the equation for x in the form of a series:
+
+    .. math::
+        g(y) = a + \sum_{n = 1}^{oo} \frac{g_n (y - f(a))^n}{n!}
+
+    where
+
+    .. math::
+        g_n = \lim_{y \to a} \left[ \frac{d^{n-1}}{dx^{n-1}}
+        \left( \frac{x-a}{f(x)-f(a)} \right) ^n \right]
+
+    The theorem further states that this series has a non-zero radius of convergence,
+    i.e., x represents an analytic function of y in a neighbourhood of y = f(a).
+
+    Parameters
+    ==========
+
+    f : expression, the expansion of whose inverse is to be found.
+
+    x : variable, in terms of which equation is given, and in terms of which, the
+    expansion will be provided.
+
+    a : point on x-axis in the neighbourhood of which, the inverse expansion is to be found.
+
+    n : no of terms required in the expansion; default value is 3 terms.
+
+    Examples
+    ========
+
+    >>> from sympy.series.series import inversion_series
+    >>> from sympy import exp
+    >>> from sympy.abc import x
+
+    Let the equation whose inverse's expansion is to be found be y = f(x)
+    then the first argument to be supplied must be f(x).
+
+    >>> f = exp(x)
+
+    In this case, the inverse series is that of the logarithmic function around 1 = f(0):
+
+    >>> inversion_series(f,x,0,4)
+    x + (x - 1)**3/3 - (x - 1)**2/2 - 1
+
+    Functions with vanishing derivative cannot be inverted with this method
+    >>> f = x**2
+    >>> try:
+    ...    inversion_series(f,x,0,2)
+    ... except ValueError as ex:
+    ...     print(repr(ex))
+    ValueError('The inversion formula requires non-zero derivative.')
+
+    Note
+    ====
+
+    There is a straightforward derivation using complex analysis and contour integration;
+    the complex formal power series version is a consequence of knowing the formula for polynomials,
+    so the theory of analytic functions may be applied. Actually, the machinery from analytic
+    function theory enters only in a formal way in this proof, in that what is really needed is
+    some property of the formal residue, and a more direct formal proof is available.
```

Comment:
This is not relevant content for the docstring. There is a link to wikipedia and other pages that give more information. This docstring should just give the mathematical definition of what is being returned but otherwise should just explain how to use the function and any relevant points about what it can and can't do.